projects/packages/masterbar/src/admin-menu/class-base-admin-menu.php

```diff
@@ -0,0 +1,798 @@
+<?php
+/**
+ * Base Admin Menu file.
+ *
+ * @package jetpack-masterbar
+ */
+
+namespace Automattic\Jetpack\Masterbar;
+
+use Automattic\Jetpack\Status;
+
+/**
+ * Class Base_Admin_Menu
+ */
+abstract class Base_Admin_Menu {
+	/**
+	 * Holds class instances.
+	 *
+	 * @var array
+	 */
+	protected static $instances;
+
+	/**
+	 * Whether the current request is a REST API request.
+	 *
+	 * @var bool
+	 */
+	protected $is_api_request = false;
+
+	/**
+	 * Domain of the current site.
+	 *
+	 * @var string
+	 */
+	protected $domain;
+
+	/**
+	 * The CSS classes used to hide the submenu items in navigation.
+	 *
+	 * @var string
+	 */
+	const HIDE_CSS_CLASS = 'hide-if-js';
+
+	/**
+	 * Identifier denoting that the default WordPress.com view should be used for a certain screen.
+	 *
+	 * @var string
+	 */
+	const DEFAULT_VIEW = 'default';
+
+	/**
+	 * Identifier denoting that the classic WP Admin view should be used for a certain screen.
+	 *
+	 * @var string
+	 */
+	const CLASSIC_VIEW = 'classic';
+
+	/**
+	 * Identifier denoting no preferred view has been set for a certain screen.
+	 *
+	 * @var string
+	 */
+	const UNKNOWN_VIEW = 'unknown';
+
+	/**
+	 * Base_Admin_Menu constructor.
+	 */
+	protected function __construct() {
+		$this->is_api_request = defined( 'REST_REQUEST' ) && REST_REQUEST || isset( $_SERVER['REQUEST_URI'] ) && str_starts_with( filter_var( wp_unslash( $_SERVER['REQUEST_URI'] ) ), '/?rest_route=%2Fwpcom%2Fv2%2Fadmin-menu' );
+		$this->domain         = ( new Status() )->get_site_suffix();
+
+		add_action( 'admin_menu', array( $this, 'reregister_menu_items' ), 99998 );
+		add_action( 'admin_menu', array( $this, 'hide_parent_of_hidden_submenus' ), 99999 );
+
+		if ( ! $this->is_api_request ) {
+			add_filter( 'admin_menu', array( $this, 'override_svg_icons' ), 99999 );
+			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ), 11 );
+			add_action( 'admin_head', array( $this, 'set_site_icon_inline_styles' ) );
+			add_action( 'in_admin_header', array( $this, 'add_dashboard_switcher' ) );
+			add_action( 'admin_footer', array( $this, 'dashboard_switcher_scripts' ) );
+			add_action( 'admin_menu', array( $this, 'handle_preferred_view' ), 99997 );
+			add_filter( 'admin_body_class', array( $this, 'admin_body_class' ) );
+
+			// Do not inject core mobile toggle when the user wants to use the WP Admin interface.
+			if ( ! $this->use_wp_admin_interface() ) {
+				add_action( 'adminmenu', array( $this, 'inject_core_mobile_toggle' ) );
+			}
+		}
+	}
+
+	/**
+	 * Returns class instance.
+	 *
+	 * @return Admin_Menu
+	 */
+	public static function get_instance() {
+		$class = static::class;
+
+		if ( empty( static::$instances[ $class ] ) ) {
+			static::$instances[ $class ] = new $class();
```

Comment:
May as well just suppress the issue here. Technically it's accurate, but I think we're safe in assuming that if someone calls `Admin_Menu_Base::get_instance()` they deserve what they get.
```suggestion
			// @phan-suppress-next-line PhanTypeInstantiateAbstract -- If someone calls `Admin_Menu_Base::get_instance()` they deserve what they get.
			static::$instances[ $class ] = new $class();
```